TestModels/SimpleTypes/SimpleDouble/codegen-patches/rust/dafny-4.5.0.patch

```diff
@@ -0,0 +1,1762 @@
+index 0609bbe6..5a759dc2 100644
+--- b/TestModels/SimpleTypes/SimpleDouble/runtimes/rust/dafny_impl/src/implementation_from_dafny.rs
++++ a/TestModels/SimpleTypes/SimpleDouble/runtimes/rust/dafny_impl/src/implementation_from_dafny.rs
+@@ -1,7 +1,7 @@
+ #![allow(warnings, unconditional_panic)]
+ #![allow(nonstandard_style)]
+ pub use dafny_standard_library::implementation_from_dafny::*;
+-
++    
+ pub mod r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes {
+     #[derive(PartialEq, Clone)]
+     pub enum DafnyCallEvent<I: ::dafny_runtime::DafnyType, O: ::dafny_runtime::DafnyType> {
+@@ -108,10 +108,7 @@ pub mod r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes {
+     }
+ 
+     impl GetDoubleInput {
+-        pub fn value(
+-            &self,
+-        ) -> &::std::rc::Rc<super::r#_Wrappers_Compile::Option<::dafny_runtime::Sequence<u8>>>
+-        {
++        pub fn value(&self) -> &::std::rc::Rc<super::r#_Wrappers_Compile::Option<::dafny_runtime::Sequence<u8>>> {
+             match self {
+                 GetDoubleInput::GetDoubleInput { value } => value,
+             }
+@@ -142,7 +139,7 @@ pub mod r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes {
+     }
+ 
+     impl Eq for GetDoubleInput {}
+-
++/*
+     impl ::std::hash::Hash for GetDoubleInput {
+         fn hash<_H: ::std::hash::Hasher>(&self, _state: &mut _H) {
+             match self {
+@@ -150,7 +147,7 @@ pub mod r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes {
+             }
+         }
+     }
+-
++*/
+     impl ::std::default::Default for GetDoubleInput {
+         fn default() -> GetDoubleInput {
+             GetDoubleInput::GetDoubleInput {
+@@ -173,10 +170,7 @@ pub mod r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes {
+     }
+ 
+     impl GetDoubleOutput {
+-        pub fn value(
+-            &self,
+-        ) -> &::std::rc::Rc<super::r#_Wrappers_Compile::Option<::dafny_runtime::Sequence<u8>>>
+-        {
++        pub fn value(&self) -> &::std::rc::Rc<super::r#_Wrappers_Compile::Option<::dafny_runtime::Sequence<u8>>> {
+             match self {
+                 GetDoubleOutput::GetDoubleOutput { value } => value,
+             }
+@@ -207,7 +201,7 @@ pub mod r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes {
+     }
+ 
+     impl Eq for GetDoubleOutput {}
+-
++/*
+     impl ::std::hash::Hash for GetDoubleOutput {
+         fn hash<_H: ::std::hash::Hasher>(&self, _state: &mut _H) {
+             match self {
+@@ -215,7 +209,7 @@ pub mod r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes {
+             }
+         }
+     }
+-
++*/
+     impl ::std::default::Default for GetDoubleOutput {
+         fn default() -> GetDoubleOutput {
+             GetDoubleOutput::GetDoubleOutput {
+@@ -302,6 +296,19 @@ pub mod r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes {
+                 ::std::rc::Rc<super::r#_simple_dtypes_dsmithydouble_dinternaldafny_dtypes::Error>,
+             >,
+         >;
++        fn GetDoubleKnownValueTest(
```

Comment:
This shouldn't be here since the smithy model doesn't actually have a `GetDoubleKnownValueTest` yet